thirdparty/faiss/faiss/utils/distances.cpp

```diff
@@ -75,8 +77,10 @@ void fvec_norms_L2sqr(
         size_t d,
         size_t nx) {
 #pragma omp parallel for if (nx > 10000)
-    for (int64_t i = 0; i < nx; i++)
-        nr[i] = fvec_norm_L2sqr(x + i * d, d);
+    for (int64_t i = 0; i < nx; i++) {
+        float norm = fvec_norm_L2sqr(x + i * d, d);
+        nr[i] = (norm == 0.0 ? 1.0 : norm);
```

Comment:
no, because it might use norm for distance = x^2 - 2xy + y^2 calculation, not for a cosine-related code